File: demo/hello_go/main_test.go
package main

import (
	"testing"
	"github.com/example/everything/libs/go"
)

func TestFormatGreeting(t *testing.T) {
	result := go_lib.FormatGreeting("test")
	expected := "Hello, test from Go!"
	if result != expected {
		t.Errorf("Expected %q, got %q", expected, result)
	}
}

func TestGetVersion(t *testing.T) {
	version := go_lib.GetVersion()
	if version == "" {
		t.Error("Version should not be empty")
	}
}

File: demo/hello_scala/MainTest.scala
package hello.scala

import org.scalatest.funsuite.AnyFunSuite
import libs.scala.Utils

class MainTest extends AnyFunSuite {
  test("formatGreeting should return correct message") {
    val message = Utils.formatGreeting("test")
    assert(message == "Hello, test from Scala!")
  }
  
  test("getVersion should return version string") {
    val version = Utils.getVersion
    assert(version == "1.0.0")
  }
}
